### ISSUE TITLE
Can't load data correctly when table is being altered by a "kafka to bigquery" dataflow job

### ISSUE THREAD

**Issue by Mpal7** [GoogleCloudDataproc/spark-bigquery-connector#473]
Hello, i've noticed that the connector doesn't seem to load data at all whenever it's being constantly altered by a dataflow job. Is this intended? if i try a simple `print(df.count(),df.show())` when the job is active i get 0 rows and no data, as soon as i stop the job the row count is greater than 0 and the `show()` actually displayes some data.

This behavior happens even if a limit the query to a certain timespan by using .option('filter',....) criteria.

Is this intended in such streaming scenarios? this is a pretty big problem since i need to run a script live on data that is being constantly updated by DataFlow.

**Comment by Mpal7** [comment]
No, the dataflow job is not truncating the existing table, older data are present indeed.  
The problem however is about the data that are currently being written from Kafka to BQ. they appear in the table (already wrote) since i can see them in the table, but i can't load them into a DF using the connector.  
How does this lag you are referring to concerns fetching data already written in the table in BQ from the dataflow job? even so with a dataflow job running for hours, datas from hours ago (of course written in the table) are not visibile. I don't think it's a question of time, is it clearer now? thank you again

**Comment by emkornfield** [comment]
> How does this lag you are referring to concerns fetching data already written in the table in BQ from the dataflow job? 

There is a lag in time between when the data written by dataflow will be visible from the BQ Read API which is what spark uses to load the data.  The time is variable, I though in general should be in the order of minutes.  So for example:
1.  Dataflow writes to BQ table at Time X
2.  Spark Reading from BQ at time X will only see data as recent as time X - Y (where per above Y is variable but should be minutes).

We are working on making Y=0

Could you expand on these two statements:
1. "No, the dataflow job is not truncating the existing table, older data are present indeed."
2. "even so with a dataflow job running for hours, datas from hours ago (of course written in the table) are not visibile."  

How long does it take the data to show-up?  How are  you measuring the time delta?
